src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/timeseries/ComputedTilesToComputedTrends.scala

```diff
@@ -0,0 +1,45 @@
+package com.microsoft.partnercatalyst.fortis.spark.analyzer.timeseries
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{ComputedTile, ComputedTrend}
+import org.apache.commons.math3.stat.regression.SimpleRegression
+import org.apache.spark.rdd.RDD
+
+object ComputedTilesToComputedTrends {
+
+  def apply(computedTiles: RDD[ComputedTile]): RDD[ComputedTrend] = {
+    val groupedTiles = computedTiles.groupBy(tileGroup)
+    groupedTiles.flatMap[ComputedTrend]((group) => {
+      val tiles = group._2.toSeq
+
+      val regressionResult = linearRegression(tiles)
+
+      val score = regressionResult.getSlope
+      if (score.isNaN) Seq[ComputedTrend]()
+      else {
+        val latestTile = tiles.sortBy(t => -t.periodstartdate).head
+        val filteredTiles = tiles.filter(t => t.period == latestTile.period)
+        val trends = filteredTiles.map(new ComputedTrend(_, score))
+        trends
+      }
+    })
+  }
+
+  def tileGroup(computedTile: ComputedTile): String = {
+    (
+      computedTile.pipelinekey,
```

Comment:
You make a reasonable point, but I wanted to factor this out to have a test to illustrate what the groups look like:

```
  it should "format groups" in {
    val group = ComputedTilesToComputedTrends.tileGroup(
      ComputedTile(2017, 0, PeriodType.Year.periodTypeName, "2017", "twitter", 10, 1, 2, "HamillHimself", 150, 0, "", Seq(), 0, Tuple3(Some("europe"), None, None))
    )
    assert(group == "(twitter,(Some(europe),None,None),10)")
  }
```

Happy to remove this if you feel  strongly about it.